app/scripts/metamask-controller.js

```diff
@@ -592,8 +593,23 @@ module.exports = class MetamaskController extends EventEmitter {
     primaryKeyring.serialize()
     .then((serialized) => {
       const seedWords = serialized.mnemonic
-      this.configManager.setSeedWords(seedWords)
-      cb(null, seedWords)
+
+      primaryKeyring.getAccounts()
+        .then((accounts) => {
+          if (accounts.length < 1) {
+            return cb(new Error('MetamaskController - No accounts found'))
+          }
+
+          seedPhraseVerifier.verifyAccounts(accounts, seedWords)
```

Comment:
I would prefer this were broken out into its own method on the controller, and was called on:
- Account creation
- Vault unlock

Reply:
Again, I think this would be much more useful if it were also called during account creation and vault unlock. Only doing it during `placeSeedWords` means it will only trigger on the very first account, or when someone reveals the seed phrase, which are two very infrequent events.